### ISSUE TITLE
trigEngine = 'ecma' consumes an extra character before the trigger

### ISSUE THREAD

**Issue by bhataktaBhai** [L3MON4D3/LuaSnip#947]
The snippet `s({ trig = 'hi', trigEngine = 'ecma', wordTrig = false }, t('hello'))` consumes exactly one extra character before the 'hi' if present
- `hi<tab>` -> `hello`
- `hhi<tab>` -> `hello`
- `hhhi<tab>` -> `hhello`

I checked that the consumed character is part of the trigger by defining
```lua
s(
   { trig = 'hi', trigEngine = 'ecma', wordTrig = false },
   f(function (_, snip)
      return 'hello:' .. snip.trigger
   end)
)
```
This behaves as
- `hi<tab>` -> `hello:hi`
- `hhi<tab>` -> `hello:hhi`
- `hhhi<tab>` -> `hhello:hhi`

Changing the trigEngine to `'pattern'` works as expected in all cases.

**Comment by bhataktaBhai** [comment]
Changing `return line_to_cursor:sub(match.begin_ind - 1), match.groups` to `return line_to_cursor:sub(match.begin_ind), match.groups` on line 54 in luasnip.nodes.util.trig_engines makes things work as expected.

Was there any reason for the - 1 or is it just a mistake?

**Comment by L3MON4D3** [comment]
hard problem #<ignorethispls></ignorethispls>4
fixed with a658ae2
Thanks for the report!
